client/my-sites/plan-features-2023-grid/header-price.tsx

```diff
@@ -1,44 +1,57 @@
-import { isWpcomEnterpriseGridPlan } from '@automattic/calypso-products';
-import PropTypes from 'prop-types';
+import { isWpcomEnterpriseGridPlan, PlanSlug } from '@automattic/calypso-products';
+import { useSelector } from 'react-redux';
 import PlanPrice from 'calypso/my-sites/plan-price';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import usePlanPrices from '../plans/hooks/use-plan-prices';
+import { PlanProperties } from './types';
 
 interface PlanFeatures2023GridHeaderPriceProps {
-	planName: string;
-	discountPrice: number | null;
-	currencyCode: string | null;
-	rawPrice: number;
+	planProperties: PlanProperties;
 	is2023OnboardingPricingGrid: boolean;
 	isLargeCurrency: boolean;
 }
 
 const PlanFeatures2023GridHeaderPrice = ( {
-	planName,
-	discountPrice,
-	currencyCode,
-	rawPrice,
+	planProperties,
 	is2023OnboardingPricingGrid,
 	isLargeCurrency,
 }: PlanFeatures2023GridHeaderPriceProps ) => {
+	const { planName, showMonthlyPrice } = planProperties;
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const planPrices = usePlanPrices( {
+		planSlug: planName as PlanSlug,
+		monthly: showMonthlyPrice,
+	} );
+	const shouldShowDiscountedPrice = Boolean(
+		planPrices.planDiscountedRawPrice || planPrices.discountedRawPrice
+	);
+	const discountedPricing = {
+		crossoutPrice: shouldShowDiscountedPrice ? planPrices.rawPrice : null,
+		price: shouldShowDiscountedPrice
+			? planPrices.planDiscountedRawPrice || planPrices.discountedRawPrice
+			: planPrices.rawPrice,
+	};
+
 	if ( isWpcomEnterpriseGridPlan( planName ) ) {
 		return null;
 	}
 
 	return (
 		<div className="plan-features-2023-grid__pricing">
-			{ discountPrice && (
+			{ shouldShowDiscountedPrice && (
 				<span className="plan-features-2023-grid__header-price-group">
 					<div className="plan-features-2023-grid__header-price-group-prices">
 						<PlanPrice
 							currencyCode={ currencyCode }
-							rawPrice={ rawPrice }
+							rawPrice={ discountedPricing.crossoutPrice ?? undefined }
```

Comment:
Since we are already checking for `shouldShowDiscountedPrice` in the component, maybe the `discountedPricing` object is redundant? This could be:
```suggestion
							rawPrice={ planPrices.rawPrice }
```

Reply:
Good call, @aneeshd16. Yes, it is redundant. I think the intention with keeping it separate was to more easily introduce conditioning needed for the cost savings (where the refactors came from). but we'll cross that line when we get to it.

Reply:
Updated. Thanks for the great questions/suggestions.
Let's do another round of testing maybe jic 😬